wordpress-mobile/WordPress-Android
# UI Modernization Posts: Update post list layout

Fixes #19341 

This PR is the second in a series dedicated to updating the post list layout. This PR focuses on updating to the new design for the list. See eYeHXEMDbnFptE40xUqZ2T-fi-951%3A43566 for design details

**Includes**
- Remove buttons from row
~~Change postInfo from List<UiString> to List<PostInfo>~~
~~Move info labels out of statuses to the postInfo line~~
- Adjust the title and excerpt logic around untitled posts. It is possible to create a post with no title and no excerpt, so Untitled should show in this case
- Hide/show author filter when search is expanded/collapsed
- Update unit tests

-----------
The following decision were made here on 23 Oct 2023 p1697809073777199-slack-C04SFL0RP51
- Reapply the existing production logic for showing info labels in the bottom line
- Reapply the existing production logic for showing status labels in the top line when search

-----------
Not Included
* This PR does not include any context menu changes. They will be handled in the next PR of the series.
* This PR will need a design review

**Merge Instructions**
- Ensure #19424 has been merged
- Make sure the target branch is `feature/ui-modernization-posts-and-pages`
- Remove the Do Not Merge label
- Merge as normal

Published Dark | Published Light
-- | --
<img width="250" height="500" alt="Alt desc" src="https://github.com/wordpress-mobile/WordPress-Android/assets/506707/bdc6c73f-e298-442a-b7f6-2102ee6a215d"> | <img width="250" height="500" alt="Alt desc" src="https://github.com/wordpress-mobile/WordPress-Android/assets/506707/23caf37c-2890-42d4-bf43-646bdd5dac3b">

**Example set one:**
Row 1: Post with a title, no excerpt, and no featured image
Row 2: Post with no title, no excerpt, and no featured image
Row 3: Post with a title that takes up 3 lines, no excerpt shown, and a featured image
Row 4: Post with a title, excerpt, and a featured image
Row 5: Post with no title, an excerpt, and no featured image.

Published Dark | Published Light
-- | --
<img width="250" height="500" alt="Alt desc" src="https://github.com/wordpress-mobile/WordPress-Android/assets/506707/350d601f-f628-4e24-8c68-6e0a306e2ccc"> | <img width="250" height="500" alt="Alt desc" src="https://github.com/wordpress-mobile/WordPress-Android/assets/506707/30b4c2c0-106f-4677-9818-5d2ad352ed11">

**Example set two:**
Row 1: Post with a title, excerpt, and blocks that don't translate to text and/or empty lines between text, a status message, no featured image
Row 2: Post with title, excerpt, featured image, and two info labels (private + sticky)
Row 3: Title, no excerpt, and a featured image

Search Dark | Search Light
-- | --
<img width="250" height="500" alt="Alt desc" src="https://github.com/wordpress-mobile/WordPress-Android/assets/506707/eb5779b9-5188-4383-afb2-16cc2c2285fe"> | <img width="250" height="500" alt="Alt desc" src="https://github.com/wordpress-mobile/WordPress-Android/assets/506707/61682e31-d105-46e5-a3eb-04becfb8c998">

**Example set three:**
Row 1: Post with a title that >= 3 lines, no excerpt, featured image, and published label on top line
Row 2: Post with no title, excerpt, no featured image, and two info labels (private + sticky)
Row 3: Post with a title, excerpt, blocks that don't translate to text and/or empty lines between text, a status message. no featured image, and published label on to top line 
Row 4: Post with title, excerpt, featured image, two info labels (private + sticky), and private label on top line

**To test:**
- Install the Jetpack app
- Login 
- Select a site
- Navigate to My Site > Post
- ✅ Verify the loading view shows when the list is loading and that there is placeholder for the image
- ✅ Verify the post list is shown and you can navigate through all the tabs
- ✅ Verify the design matches the figma design: eYeHXEMDbnFptE40xUqZ2T-fi-836%3A5909 and the example sets above
- ✅ Verify the "more" button launches when tapped and you can navigate as expected
- ✅ Verify the bottom action buttons are not shown


## Regression Notes
1. Potential unintended areas of impact
The post list doesn't work as expected

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual and unit testing

3. What automated tests I added (or what prevented me from doing so)
`PostListItemUiStateHelperTest` and view model tests

PR submission checklist:

UI Changes testing checklist:
